src/bosh_azure_cpi/lib/cloud/azure/disk/config_disk_manager.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+module Bosh::AzureCloud
+  class ConfigDiskManager
+    include Helpers
+
+    BOSH_CONFIG_DISK_LABEL = 'azure_cfg_dsk'
+    CONFIG_DISK_CONTAINER  = 'azure-config-disks'
+    CONFIG_DISK_MOUNT_POINT = 'azure_config_disk_mount'
+    CONFIG_DISK_FILE_PATH_PREFIX = 'azure_config_disk_image'
+    CONFIGS_RELATIVE_PATH = 'configs'
+    CONFIG_DISK_SIZE = 24 # in MB, azure have a requirement for the min size.
+    MEGA_SIZE = 1024 * 1024
+    def initialize(blob_manager, disk_manager2, storage_account_manager)
+      @blob_manager = blob_manager
+      @disk_manager2 = disk_manager2
+      @storage_account_manager = storage_account_manager
+      @logger = Bosh::Clouds::Config.logger
+    end
+
+    # Returns the config disk uri.
+    def prepare_config_disk(resource_group_name, vm_name, location, metadata_obj, user_data_obj)
+      disk = create_config_disk(resource_group_name, vm_name, location, metadata_obj, user_data_obj)
+      disk
+    end
+
+    private
+
+    def create_config_disk(resource_group_name, vm_name, location, meta_data_obj, user_data_obj)
+      mounted_dir = nil
+      config_disk_file_path = nil
+      config_disk_vhd_file_path = nil
+      disk_name = nil
+      page_blob_created = false
+      command_runner = CommandRunner.new
+      umounted = false
+      begin
+        config_disk_file = Tempfile.new(CONFIG_DISK_FILE_PATH_PREFIX)
+        config_disk_file.close
+        config_disk_file_path = config_disk_file.path
+
+        mk_image_file_cmd = "dd if=/dev/zero of=#{config_disk_file_path} bs=#{MEGA_SIZE} count=#{CONFIG_DISK_SIZE}"
+        command_runner.run_command(mk_image_file_cmd)
+
+        mkfs_cmd = "mkfs.ext4 -F #{config_disk_file_path} -L #{BOSH_CONFIG_DISK_LABEL}"
+        command_runner.run_command(mkfs_cmd)
+
+        mounted_dir = Dir.mktmpdir(CONFIG_DISK_MOUNT_POINT)
```

Comment:
mktmpdir not always generate uniq dir. see [issue](https://github.com/Albacore/albacore/issues/101).
don't use it here.

Reply:
we have the prefix, and the folder will be cleaned up at the end of this method.
so i think use it here is ok.

Reply:
say if there are 10 VMs creating at the same time?

Reply:
cool, will generate uuid as prefix then.